kik_unofficial/client.py

```diff
@@ -914,15 +896,31 @@ def split_elements(data_bytes):
             b'<k .*?>',
             b'<pong/>',
             b'<iq .*?>.*?</iq>',
-            b'<message .*?>.*?</message>',
-            b'<stc .*?>.*?</stc>'
+            b'<stc .*?>.*?</stc>',
+            b'<ack .*?/>'
         ]
-
-        # Extract all matching elements
         elements = []
         for pattern in element_patterns:
-            matches = re.findall(pattern, data_bytes)
+            matches = re.findall(pattern, data_bytes, re.DOTALL)
             elements.extend(matches)
+            for match in matches:
+                data_bytes = data_bytes.replace(match, b'', 1)
+
+        complete_message_pattern = b'<message .*?>.*?</message>'
```

Comment:
If you split the elements and then try identify multi-packets, I think it is going to be error prone. It will be better to first make sure we have the complete XML, and then parse it, like you did previosuly.